cloudfoundry/resource_cf_route_service_binding.go

```diff
@@ -63,48 +94,106 @@ func resourceRouteServiceBindingCreate(ctx context.Context, d *schema.ResourceDa
 	}
-	routes, _, err := session.ClientV2.GetServiceBindingRoutes(serviceID)
+
+	options := client.NewServiceRouteBindingListOptions()
+	options.ServiceInstanceGUIDs = client.Filter{Values: []string{serviceID}}
+	options.RouteGUIDs = client.Filter{Values: []string{routeID}}
+
+	routeBinding, err := session.ClientGo.ServiceRouteBindings.Single(context.Background(), options)
+
 	if err != nil {
 		return diag.FromErr(err)
 	}
-	found := false
-	for _, route := range routes {
-		if route.GUID == routeID {
-			found = true
-			break
+
+	d.SetId(routeBinding.GUID)
+	return nil
+}
+
+func resourceRouteServiceBindingRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	session := meta.(*managers.Session)
+
+	routeServiceBinding, err := session.ClientGo.ServiceRouteBindings.Get(context.Background(), d.Id())
+
+	if err != nil {
+		if strings.Contains(err.Error(), "CF-ResourceNotFound") {
+			d.SetId("")
+			return nil
 		}
+		return diag.Errorf("Error when reading routeServiceBinding with id '%s': %s", d.Id(), err)
 	}
-	if !found {
-		d.SetId("")
-		return diag.Errorf("Route '%s' not found in service instance '%s'", routeID, serviceID)
-	}
 
-	d.Set("service_instance", serviceID)
-	d.Set("route", routeID)
+	d.Set("service_instance", routeServiceBinding.Relationships.ServiceInstance.Data.GUID)
+	d.Set("route", routeServiceBinding.Relationships.Route.Data.GUID)
+
 	return nil
 }
 
-func resourceRouteServiceBindingDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+func resourceRouteServiceBindingDelete(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	session := meta.(*managers.Session)
 
-	serviceID := d.Get("service_instance").(string)
-	routeID := d.Get("route").(string)
-	_, err := session.ClientV2.DeleteServiceBindingRoute(serviceID, routeID)
+	jobGUID, err := session.ClientGo.ServiceRouteBindings.Delete(context.Background(), d.Id())
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if jobGUID != "" {
+		err = session.ClientGo.Jobs.PollComplete(context.Background(), jobGUID, nil)
+	}
 	return diag.FromErr(err)
 }
+
+func upgradeStateRouteServiceBindingStateV0toV1ChangeID(ctx context.Context, rawState map[string]any, meta any) (map[string]any, error) {
+	session := meta.(*managers.Session)
+
+	if len(rawState) == 0 {
+		log.Println("[DEBUG] Empty RouteServiceBinding; nothing to migrate.")
+		return rawState, nil
+	}
+
+	log.Printf("[DEBUG] Attributes before migration: %#v", rawState)
+	options := client.NewServiceRouteBindingListOptions()
+	options.ServiceInstanceGUIDs = client.Filter{Values: []string{rawState["service_instance"].(string)}}
+	options.RouteGUIDs = client.Filter{Values: []string{rawState["route"].(string)}}
+
+	routeBinding, err := session.ClientGo.ServiceRouteBindings.Single(context.Background(), options)
+
+	if err != nil {
+		log.Println("[DEBUG] Failed to migrate RouteServiceBinding id: did not find the route service binding.")
```

Comment:
Maybe ignore error and set id to "" if the error is a 404 error

Reply:
Agree and disagree? Maybe? It's debatable whether we should limit the state upgrader's responsibility to just upgrading the state and not modifying anything else.